src/buildbot.py

```diff
@@ -37,6 +37,15 @@
 BUILDBOT_REVISION = os.environ.get('BUILDBOT_REVISION', None)
 BUILDBOT_BUILDNUMBER = os.environ.get('BUILDBOT_BUILDNUMBER', None)
 BUILDBOT_BUILDERNAME = os.environ.get('BUILDBOT_BUILDERNAME', None)
+BUILDBOT_MASTERNAME = os.environ.get('BUILDBOT_MASTERNAME', None)
+
+# Possible masters include None (running locally), the waterfall integration
+# bot, or the emscripten-releases bot.
+WATERFALL_BOT = 'client.wasm.llvm'
+EMSCRIPTEN_RELEASES_BOT = 'emscripten-releases'
+
+assert BUILDBOT_MASTERNAME in [None, WATERFALL_BOT, EMSCRIPTEN_RELEASES_BOT], \
```

Comment:
I realized we'll also need to pass the info about whether the build is a CI build or a try build, and only try to upload from CI builds. I added `BUILDBOT_BUCKET` to the env (which IIUC should be either "ci" or "try").

Reply:
Thanks, I'll open a PR.